lib/types.js

```diff
@@ -0,0 +1,88 @@
+"use strict";
+
+const conversions = require("webidl-conversions");
+
+const utils = require("./utils");
+
+module.exports.generateTypeConversion = function (name, idlType, argAttrs, customTypes, opts) {
```

Comment:
This function is very long. I think it would benefit from calling out to a few non-exported factored-out functions for the two cases of sequence and IDL type. What do you think?